contracts/VaultV3.vy

```diff
@@ -630,11 +625,19 @@ def _process_report(strategy: address) -> (uint256, uint256):
         self.strategies[strategy].total_loss += loss
         self.strategies[strategy].current_debt -= loss
 
-        locked_profit_before_loss: uint256 = self._calculate_locked_profit()
-        if locked_profit_before_loss > loss:
-            self.locked_profit = locked_profit_before_loss - loss
+        if self.profit_buffer <= loss:
+            # If loss is too big for the profit buffer, we set buffer to zero and delete all profits
+            self.total_debt -= loss - self.profit_buffer
+            self.profit_buffer = 0
+            self.profit_history = []
         else:
-            self.locked_profit = 0
+            loss_factor: uint256 = MAX_BPS - (loss * MAX_BPS / self.profit_buffer)
+            profit_history_updated: DynArray[Profit, MAX_PROFITS_HISTORY] = []
+            for e in self.profit_history:
```

Comment:
I see your point, and it makes sense. I haven't done it like that because then it wouldn't follow the economics proposed, where all profits are affected equally and loss is distributed equally across profits. 

I'll changed it to your proposal as it optimizes the code.